packages/amplify_datastore_plugin_interface/test/amplify_modelschema_to_map_test.dart

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:amplify_datastore_plugin_interface/amplify_datastore_plugin_interface.dart';
+
+import 'package:flutter_test/flutter_test.dart';
+
+import './testData/index.dart';
+
+void main() {
+  test('Blog codegen model generates modelschema with proper fields', () async {
+    ModelSchema blogSchema = Blog.schema;
+    Map<String, dynamic> map = blogSchema.toMap();
+
+    expect(map, {
+      'name': "Blog",
+      'pluralName': "Blogs",
+      'authRules': const [],
+      'fields': {
+        'id': {
+          'name': "id",
+          'type': {'fieldType': 0, 'ofModelName': ""},
+          'isRequired': true,
+          'isArray': false,
+          'association': const {},
+          'authRules': const [],
+        },
+        'name': {
+          'name': "name",
+          'type': {'fieldType': 0, 'ofModelName': ""},
+          'isRequired': true,
+          'isArray': false,
+          'association': const {},
+          'authRules': const []
+        },
+        'posts': {
+          'name': "posts",
+          'type': {'fieldType': 10, 'ofModelName': "Post"},
+          'isRequired': false,
+          'isArray': true,
+          'association': const {
+            'associationType': 0,
+            'targetName': "",
+            'associatedName': "blog",
+            'associatedType': "Post"
+          },
+          'authRules': const []
+        },
+      }
+    });
+  });
+
+  test('Comment codegen model generates modelschema with proper fields',
+      () async {
+    ModelSchema commentSchema = Comment.schema;
+    Map<String, dynamic> map = commentSchema.toMap();
+
+    expect(map, {
+      'name': 'Comment',
+      'pluralName': 'Comments',
+      'authRules': [],
+      'fields': {
+        'id': {
+          'name': 'id',
+          'type': {'fieldType': 0, 'ofModelName': ''},
```

Comment:
Hm, the integers for the field types is weird. Is it not possible to use a label of some kind, instead?

Reply:
It's not possible, we're just creating a simple map using inline declaration for dart.